flipt-client-js/src/slim/index.ts

```diff
@@ -0,0 +1,172 @@
+import init, { Engine } from '../../dist/flipt_engine_wasm_js.js';
+import { BaseFliptClient } from '../core/base';
+import { ClientOptions, ErrorStrategy } from '../core/types';
+
+export * from '../core/types';
+export * from '../core/base';
+
+export interface WasmOptions {
+  /**
+   * The WASM module to use for evaluation.
+   * Can be provided as:
+   * - A URL string pointing to the WASM file
+   * - A filesystem path to the WASM file (in Node.js environments)
+   * - An ArrayBuffer or Uint8Array containing the WASM binary
+   * - A WebAssembly.Module object
+   * 
+   * The simplest way to provide the WASM module is to import it directly from the package:
+   * ```
+   * import wasm from '@flipt-io/flipt-client-js/flipt.wasm';
+   * 
+   * const client = await FliptClient.init({ ... }, { wasm });
+   * ```
+   */
+  wasm: ArrayBuffer | Uint8Array | WebAssembly.Module | string;
+}
+
+export class FliptClient extends BaseFliptClient {
+  private updateInterval?: NodeJS.Timeout;
+
+  /**
+   * Initialize the client
+   * @param options - optional client options
+   * @param wasmOptions - options for loading WASM
+   * @returns {Promise<FliptClient>}
+   */
+  static async init(
+    options: ClientOptions = {
+      namespace: 'default',
+      url: 'http://localhost:8080',
+      reference: '',
+      updateInterval: 120,
+      errorStrategy: ErrorStrategy.Fail
+    },
+    wasmOptions?: WasmOptions
+  ): Promise<FliptClient> {
+    if (!wasmOptions || !wasmOptions.wasm) {
+      throw new Error('WASM module must be provided in slim mode. Use the standard client or provide a wasm module.');
+    }
+
+    const namespace = options.namespace ?? 'default';
+
+    let url = options.url ?? 'http://localhost:8080';
+    url = url.replace(/\/$/, '');
+    url = `${url}/internal/v1/evaluation/snapshot/namespace/${namespace}`;
+
+    if (options.reference) {
+      url = `${url}?reference=${options.reference}`;
+    }
+
+    const headers: Record<string, string> = {
+      Accept: 'application/json',
+      'x-flipt-accept-server-version': '1.47.0'
+    };
+
+    if (options.authentication) {
+      if ('clientToken' in options.authentication) {
+        headers['Authorization'] =
+          `Bearer ${options.authentication.clientToken}`;
+      } else if ('jwtToken' in options.authentication) {
+        headers['Authorization'] = `JWT ${options.authentication.jwtToken}`;
+      }
+    }
+
+    let fetcher = options.fetcher;
+
+    if (!fetcher) {
+      // Check if we're in a Node.js environment
+      const isNode = typeof window === 'undefined';
```

Comment:
I believe there is no reason in this check because of ` private updateInterval?: NodeJS.Timeout;` and `node-fetch`. Probably it's impossible to run this in browser.